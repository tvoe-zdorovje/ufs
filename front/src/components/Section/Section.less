@import "../Shared";

.section {
  margin-top: 12px;
  border: 1px solid @green;
}

.header {
  padding: 1px 8px 1px 4px;
  margin: -12px 0 0 8px;
  color: @green;
  background-color: #fff;
  position: absolute;
  z-index: 10;
}

.container {
  padding: 10px 5px 5px 5px;
}
