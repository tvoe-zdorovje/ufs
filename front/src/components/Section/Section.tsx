import React from 'react'

const styles = require('./Section.less')

interface SectionProps {
  header: string,
  children?: any,
}

export default (props: SectionProps) => (
  <section className={styles.section}>
    <div className={styles.header}>{props.header}</div>
    <div className={styles.container}>
      {props.children}
    </div>
  </section>
);
