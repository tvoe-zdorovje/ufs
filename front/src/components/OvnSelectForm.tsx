import React from 'react'
import {
  Button,
  Row,
  Table
} from 'antd';
import Section from './Section/Section'
import { OVN } from 'interfaces/dto/OVN.interface'
const styles = require('./Shared.less')

const columns = [{
  title: '№ ОВН',
  dataIndex: 'num',
}, {
  title: 'Сумма',
  dataIndex: 'amount',
}, {
  title: 'Дата создания',
  dataIndex: 'createdDate',
}, {
  title: 'ИНН',
  dataIndex: 'inn',
}, {
  title: 'Наименование ЮЛ',
  dataIndex: 'legalEntityShortName',
}, {
  title: 'Представитель',
  dataIndex: 'repFio',
}];

export interface OvnSelectFormProps {
  getOvnList: Function,
  nextStepNew: any,
  nextStepOvn: any,
  account: {
    id: string,
    status: string
  }
  loaded: boolean,
  ovnList: Array<any>,
}

export default class OvnSelectForm extends React.Component<OvnSelectFormProps, any> {
  constructor(props) {
    super(props)
    this.nextStepOvn = this.nextStepOvn.bind(this)
  }

  rowSelection = {
    type: ('radio' as any),
    onChange: (key: string[], selectedRows: Array<OVN>) => {
      this.setState({ selectedRows })
    },
  }

  nextStepOvn = () => {
    this.props.nextStepOvn(this.state.selectedRows[0])
  }

  componentDidMount() {
    const { loaded, account, getOvnList } = this.props

    if (loaded) {
      return
    }
    getOvnList(account)
  }

  render() {
    const disabledOVN = !this.state || !this.state.selectedRows || this.state.selectedRows.length === 0
    return (
      <div>
        <Section header="Объявления на взнос наличными">
          <Table
            rowKey="id"
            rowSelection={this.rowSelection}
            columns={columns}
            dataSource={this.props.ovnList}
            locale={{
              filterConfirm: 'ОК',
              filterReset: 'Сбросить',
              emptyText: 'Данные отсутствуют',
            }}
          />
        </Section>
        <Row className={styles.rightAligned}>
          <Button className={styles.greenButton} onClick={this.props.nextStepNew}>
            Создать новую операцию
          </Button>
          <Button className={disabledOVN ? '' : styles.greenButton} disabled={ disabledOVN } onClick={this.nextStepOvn}>
            На основании выбранного ОВН
          </Button>
        </Row>
      </div>
    );
  }
}
