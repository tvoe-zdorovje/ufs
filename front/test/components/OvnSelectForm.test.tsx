import React from 'react'
import {
  mount,
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import {
  Table,
  Button
} from 'antd'

import OvnSelectForm, { OvnSelectFormProps } from 'components/OvnSelectForm'
import Section from 'components/Section/Section'

describe('Компонент <OvnSelectForm />', () => {
  const props: OvnSelectFormProps = {
    getOvnList: () => {
    },
    nextStepNew: () => {
    },
    nextStepOvn: () => {
    },
    account: {
      id: '1',
      status: '0'
    },
    loaded: true,
    ovnList: [],
  }
  const shallowWrapper = shallow(<OvnSelectForm {...props} />)
  const renderWrapper = render(<OvnSelectForm {...props} />)

  it('использует <Section/>', () => {
    expect(shallowWrapper.find(Section)).toHaveLength(1)
  })

  it('отображает компонент <Table/>', () => {
    expect(shallowWrapper.find(Table)).toHaveLength(1)
  })

  it('отображает 2 кнопки <Button />', () => {
    expect(shallowWrapper.find(Button)).toHaveLength(2)
  })

  it('запрашивает getOvnList если не загружено', () => {
    const spy = jest.fn()
    const getOvnListSpy = { ...props, getOvnList: spy, loaded: false }
    mount(<OvnSelectForm { ...getOvnListSpy} />)
    expect(spy).toBeCalled()
  })

  it('работает кнопка Создать новую операцию', () => {
    const spy = jest.fn()
    shallowWrapper.setProps({ nextStepNew: spy })
    shallowWrapper.update()
    shallowWrapper.find(Button).at(0).simulate('click', {
      preventDefault() {
      }
    })
    expect(spy).toBeCalled()
  })

  it('кнопка На основании выбранного ОВН disabled, если ничего не выбрано', () => {
    shallowWrapper.setState({})
    shallowWrapper.update()
    expect(shallowWrapper.find(Button).at(1).prop('disabled')).toBeTruthy()
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
