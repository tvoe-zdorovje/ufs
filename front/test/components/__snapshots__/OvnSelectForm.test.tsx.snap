// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <OvnSelectForm /> совпадает со своим снимком 1`] = `
<div>
  <section
    class="section"
  >
    <div
      class="header"
    >
      Объявления на взнос наличными
    </div>
    <div
      class="container"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-large ant-table-empty ant-table-scroll-position-left"
            >
              <div
                class="ant-table-content"
              >
                <div
                  class="ant-table-body"
                >
                  <table
                    class=""
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-selection-column"
                        >
                          <span />
                        </th>
                        <th
                          class=""
                        >
                          <span>
                            № ОВН
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span>
                            Сумма
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span>
                            Дата создания
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span>
                            ИНН
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span>
                            Наименование ЮЛ
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span>
                            Представитель
                          </span>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  class="ant-table-placeholder"
                >
                  Данные отсутствуют
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="ant-row rightAligned"
  >
    <button
      class="ant-btn greenButton"
      type="button"
    >
      <span>
        Создать новую операцию
      </span>
    </button>
    <button
      class="ant-btn"
      disabled=""
      type="button"
    >
      <span>
        На основании выбранного ОВН
      </span>
    </button>
  </div>
</div>
`;
